web/ui/react-app/src/utils/urlParams.test.ts

```diff
@@ -33,7 +34,61 @@ describe('decodePanelOptionsFromQueryString', () => {
     expect(decodePanelOptionsFromQueryString('')).toEqual([]);
   });
   it('returns and array of parsed params when query string is non-empty', () => {
-    expect(decodePanelOptionsFromQueryString(query)).toEqual(panels);
+    expect(decodePanelOptionsFromQueryString(query)).toMatchObject(panels);
+  });
+});
+
+describe('parseOption', () => {
+  it('should return empty object for invalid param', () => {
+    expect(parseOption('invalid_prop=foo')).toEqual({});
+  });
+  it('should parse expr param', () => {
+    expect(parseOption('expr=foo')).toEqual({ expr: 'foo' });
+  });
+  it('should parse stacked', () => {
+    expect(parseOption('stacked=1')).toEqual({ stacked: true });
+  });
+  it('should parse end_input', () => {
+    expect(parseOption('end_input=2019-10-25%2023%3A37')).toEqual({ endTime: moment.utc('2019-10-25 23:37').valueOf() });
+  });
+  it('should parse moment_input', () => {
+    expect(parseOption('moment_input=2019-10-25%2023%3A37')).toEqual({ endTime: moment.utc('2019-10-25 23:37').valueOf() });
+  });
+  describe('step_input', () => {
+    it('should return step_input parsed if > 0', () => {
+      expect(parseOption('step_input=2')).toEqual({ resolution: 2 });
+    });
+    it('should return empty object if step is equal 0', () => {
+      expect(parseOption('step_input=0')).toEqual({});
+    });
+  });
+  describe('range_input', () => {
+    it('should return range parsed if its ot null', () => {
+      expect(parseOption('range_input=2h')).toEqual({ range: 7200 });
+    });
+    it('should return empty object for invalid value', () => {
+      expect(parseOption('range_input=h')).toEqual({});
+    });
+  });
+  describe('Parse type param', () => {
+    it('should return "graph" if tab value is 0', () => {
+      expect(parseOption('tab=0')).toEqual({ type: 'graph' });
+    });
+    it('should return "table" if tab value is 1', () => {
+      expect(parseOption('tab=0')).toEqual({ type: 'graph' });
```

Comment:
Here you still need to adjust the values copied from the previous test to be table instead of graph.

Reply:
Better to use `PanelType.Graph` and `PanelType.Table` here instead of direct strings?